internal/component/prometheus/remotewrite/remote_write.go

```diff
@@ -94,6 +94,8 @@ func New(o component.Options, c Arguments) (*Component, error) {
 		return nil, err
 	}
 	ls := service.(labelstore.LabelStore)
+	// Enable the labelstore since it only exists to serve remote_write.
+	ls.Enable()
```

Comment:
If a config reload removes all remote_write components should we disable? I don't think this is a super important use case to consider, but wanted to discuss.

Reply:
We would have to somehow track the usage which feels  not worth it. Labelstore would need to track calls to enable and disable but I feel there could be some sharp edges there. IMO no one will be flip flopping without restarting.

Reply:
I wasn't considering this at the time, but fleet management is probably a use case where a remote_write could disappear and reappear without restart relatively easily.

Reply:
If it started with remote_write it’s fine it is already enabled. The specific case is going from no remote writes to one. Might be better to return 0 in all cases since both the scrape and remote write will see that as new values. Would need to double check a few things though

Reply:
I would be in favour of tracking usage. The call to .Enable() could be replaced with .Register() and increment a number of users of the labelstore. When number of users is/reaches zero, we keep labelstore emptied and disabled. When it's > 0, we keep it working. Doesn't seem like too complex and would avoid edge cases that can be hard to debug.